src/main/java/com/flat/odessa/entity/SinglePageCrawler.java

```diff
@@ -0,0 +1,107 @@
+package com.flat.odessa.entity;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class SinglePageCrawler implements Links {
+
+  private String url;
+
+  public SinglePageCrawler(String url) {
+    this.url = url;
+  }
+
+  @Override
+  public Iterator<URL> iterator() {
+    List<URL> urls = new ArrayList<>();
+    links(url).stream().forEach(item -> {
+      try {
+        urls.add(new URL(item));
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
+      }
+    });
+    return urls.iterator();
+  }
+
+  private Set<String> links(String url) {
+    String baseUrl = getBaseUrl(url);
+    Collection<String> links = tags(url, "a").stream()
+        .map(element -> element.attr("href"))
+        .collect(Collectors.toSet());
+
+    return getAbsolutePath(url, baseUrl, links);
+  }
+
+  private Set<String> getAbsolutePath(String parentUrl, String baseUrl,
+      Collection<String> links) {
+    String hostName = getHostName(parentUrl);
+    return links
+        .stream()
+        .map(item -> {
+          String transformedLink = item;
+          if (!item.startsWith("http")) {
+            try {
+              transformedLink = new URL(new URL(baseUrl), item).toString();
```

Comment:
@greySlon what about relative paths that start with `//`(relative to schema only)?
https://stackoverflow.com/a/21828923

Reply:
if item statrts with "//" then new host would return

Reply:
@greySlon my suggestion is to simplify this class as much and possible.
Lets use Links<String> instead of Links<URL>. Also I'd suggest to make generic `Links<T> extends Iterable<T>` and then several implementation one that lists urls as String and one that lists them as URLs, last should be dependent on the first. First should be dependent on the `Tags`